client/bugLogClientAngular.js

```diff
@@ -1,134 +1,171 @@
+        var config = {
+            // this is the location of the BugLog server that will be receiving the error reports (needs to be a REST/POST endpoint)
             listener: bugLogConfig.listener,
-            hostName: bugLogConfig.hostName,
+            //the application to tell BugLog from where the report is coming from.
             appName: appName,
-            apiKey: bugLogConfig.apiKey,
+            // If the BugLog server requires an API Key to talk to it, then here is where you put it.
+            apiKey: bugLogConfig.apiKey
+        };
 
-            notifyService: function(message) {
-                var msg = {};			// set defaults
-                if(typeof message == "string")
-                    msg.message = message;
-                msg.message = message.message;
-                msg.extraInfo = message.extraInfo || "";
-                msg.severity = message.severity || "ERROR";
-                msg.error = message.error || undefined;
-                msg.stack = message.stack || undefined;
+        var service = {
+            logBug: logBug
+        };
 
-                // get additional information (if any)
-                // get assets here for file version reference
-                var extra = {
-                    assets: BugLog.getAssets(),
-                    errorUrl: $window.location.href
-                };
-                if(msg.extraInfo) {
-                    extra += "<br><br><b>Extra Info:</b><br>";
-                    if(typeof msg.extraInfo=="string") {
-                        extra += msg.extraInfo;
-                    } else {
-                        if(window.JSON) {
-                            extra += JSON.stringify(msg.extraInfo);
-                        }
-                    }
-                }
+        return service;
+        /////////////////////
 
-                // see if we can get a stacktrace
-                var stacktrace = undefined;
-                if(typeof msg.stack !== "undefined") {
-                    stacktrace = msg.stack;
-                } else if(typeof msg.error !== "undefined") {
-                    stacktrace = (typeof msg.error.stack!=="undefined") ? msg.error.stack : $window.printStackTrace({e:msg.error}).join("\n");
-                }
-                if(typeof stacktrace !== "undefined") {
-                    extra += "<br><br><b>Stacktrace:</b><br><pre>"+stacktrace+"</pre>";
-                }
-                
+        /**
+         * Communicates with a BugLogHq Server
+         * @param {string} cause
+         * @param {object} error
+         * @param {string} severityCode
+         */
+        function logBug(cause, error, severityCode) {
+            var data = {};
+            data.message = 'Caused by: ' + cause;
+            data.exceptionMessage = cause || 'Unknown';
+            data.exceptionDetails =  buildStacktrace(error) || 'something went wrong';
+            data.severityCode = severityCode || 'EXCEPTION';
```

Comment:
Have you thought about making the default severity code a constant or something that can be configured somewhere? Alternatively having a consistent default value between client and server would also make sense. In the server, the default when severityCode is not provided is `ERROR`